lib/Runtime/Types/SimpleTypeHandler.cpp

```diff
@@ -135,6 +112,64 @@ namespace Js
         return newTypeHandler;
     }
 
+    template<size_t size>
+    PathTypeHandlerBase* SimpleTypeHandler<size>::ConvertToPathType(DynamicObject* instance)
```

Comment:
In one Speedometer test I saw a lot of SimpleTypeHandler::AddProperty/convert calls. This makes me think that SimpleTypeHandler type sharing isn't great (and so the PathTypeHandler sharing won't be great either). I'll try to find the case where this was happening, but I think there might be more to do here.

Is it the case that all function objects have their own SimpleTypeHandler?

Reply:
I'd like to see the code you're talking about. PathTypeHandler sharing will work in cases that SimpleTypeHandler doesn't -- basically, any case where we're changing the default properties/attributes of the object.

Reply:
Ah I see, this looks good! I'm excited to see the win we get!